pkg/query-service/app/http_handler.go

```diff
@@ -179,10 +179,10 @@ func writeHttpResponse(w http.ResponseWriter, data interface{}) {
 }
 func (aH *APIHandler) RegisterMetricsRoutes(router *mux.Router) {
 	subRouter := router.PathPrefix("/api/v2/metrics").Subrouter()
-	subRouter.HandleFunc("/query_range", aH.queryRangeMetricsV2).Methods(http.MethodPost)
-	subRouter.HandleFunc("/autocomplete/list", aH.metricAutocompleteMetricName).Methods(http.MethodGet)
-	subRouter.HandleFunc("/autocomplete/tagKey", aH.metricAutocompleteTagKey).Methods(http.MethodGet)
-	subRouter.HandleFunc("/autocomplete/tagValue", aH.metricAutocompleteTagValue).Methods(http.MethodGet)
+	subRouter.HandleFunc("/query_range", ViewAccess(aH.queryRangeMetricsV2)).Methods(http.MethodPost)
```

Comment:
What does ViewAccess do to this handler?

Reply:
Users need viewer permission to access these resources https://signoz.io/docs/userguide/authentication/#permission-matrix. As of now they are accessible without any authentication.